core/class/jMQTT.class.php

```diff
@@ -1529,7 +1529,41 @@ public function getBroker() {
         }
         return $this->_broker;
     }
-    
+
+    /**
+     * Check if a certificate file is used by the Broker
+     * If TLS is disabled, gracefully replaces the certificate by the default configuration value.
+     * @param string $certname Certificate file name
+     * @return boolean true if certificat is used
+     */
+    public function isCertUsed($certname) {
+        if ($certname == $this->getConf(self::CONF_KEY_MQTT_TLS_CA)) {
+            if (boolval($this->getConf(self::CONF_KEY_MQTT_TLS))) {
+                return true;
+            } else {
+                $this->setConfiguration(self::CONF_KEY_MQTT_TLS_CA, $this->getDefaultConfiguration(self::CONF_KEY_MQTT_TLS_CA));
+                $this->save();
```

Comment:
this kind of clean needs to be done in preSave

Reply:
Ok, got it let me known if I implemented it at the right place this time.
Initially I did it that way to be able to retain the certs on a broker with TLS disabled.